src/pages/messages/messages.ts

```diff
@@ -172,11 +192,28 @@ export class MessagesPage implements OnInit, OnDestroy {
     });
   }
 
-  sendMessage(): void {
+  showPicture({ target }: Event) {
+    const modal = this.modalCtrl.create(ShowPictureComponent, {
+      pictureSrc: (<HTMLImageElement>target).src
+    });
+
+    modal.present();
+  }
+
+  sendPictureMessage(file: File): void {
+    this.pictureService.upload(file).then((picture) => {
+      MeteorObservable.call('addMessage', 'picture',
+        this.selectedChat._id,
+        picture.url
+      ).zone().subscribe();
+    });
+  }
+
+  sendTextMessage(): void {
     // If message was yet to be typed, abort
     if (!this.message) return;
 
-    MeteorObservable.call('addMessage',
+    MeteorObservable.call('addMessage', 'text',
```

Comment:
The thing is that sometimes you'd like to use these values in the view, and an ENUM can only be represented using numbers, and not necessarily strings. Is it OK if I'll use `type` declaration rather than `enum` declaration?